### PR TITLE
Log playground requests

### PR DESCRIPTION
**Pull request by psrpinto**
This PR logs all requests to playground in the browser console, when in development environments:

Screenshot 2024-12-19 at 16 48 46

Ideally we would want the requests to show in the network tab of browser tools, which I think would be possible with a service worker (which would serve as proxy between the extension and playground), but getting that working would not be a quick fix, so I figured I wouldn't do that just yet.

### CODE REVIEW

File: src/ui/preview/PlaygroundHttpProxy.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	}
+}
+
+function logRequest( args: { request: PHPRequest; response: PHPResponse } ) {
+	const { request, response } = args;
+	const url = request.url;
+	const message = `[Request] ${ url }
+[Request body] ${ request.body }
+[Response] (${ response.httpStatusCode })`;
+	console.log( message, response.json );
```
Comment (ashfame) [inline src/ui/preview/PlaygroundHttpProxy.ts]:
```suggestion
	console.log( {
		type: 'API Request/Response',
		request: {
			url,
			body:
				typeof request.body === 'string'
					? ( () => {
							try {
								return JSON.parse( request.body );
							} catch {
								return request.body;
							}
					  } )()
					: request.body,
		},
		response: { status: response.httpStatusCode, body: response.json },
	} );
```

Comment (ashfame) [reply src/ui/preview/PlaygroundHttpProxy.ts]: The suggested change gets us this:

Screenshot 2024-12-19 at 16 48 46

**Review by ashfame — APPROVED** [review]
Nice! 🚀

File: src/ui/preview/PlaygroundHttpProxy.ts (hunk 1 above)
Comment (psrpinto) [reply src/ui/preview/PlaygroundHttpProxy.ts]: Love this, great change 👌

**Pull request merged**